### PR TITLE
fix(electron): sync settings from localStorage -> atom -> electron

### PR DESCRIPTION
**Pull request by pengx17**
- moved `appSettingAtom` to infra since we now have different packages that depends on it. There is no better place to fit in for now
- use atomEffect to sync setting changes to updater related configs to Electron side
- refactored how Electron reacts to updater config changes.

### CODE REVIEW

**Comment by pengx17** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #4957** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/4957?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5020** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5020?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #5018** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5018?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5020?utm_source=stack-comment).

**Comment by pengx17** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #4957** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/4957?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5019** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5019?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5020** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5020?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #5018** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5018?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5020?utm_source=stack-comment).

**Comment by JimmFly** [comment]
awesome

File: packages/common/infra/src/atom/settings.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
+import { setupGlobal } from '@affine/env/global';
 import { atom } from 'jotai';
 import { atomWithStorage } from 'jotai/utils';
+import { atomEffect } from 'jotai-effect';
+
+setupGlobal();
```
Comment (joooye34) [inline packages/common/infra/src/atom/settings.ts:6]: why call setup here? not in a app start progress.

File: packages/common/infra/src/atom/settings.ts [hunk 2]
```diff
@@ -63,15 +67,35 @@ const appSettingBaseAtom = atomWithStorage<AppSetting>('affine-settings', {
...
+  const settings = get(appSettingBaseAtom);
+  // some values in settings should be synced into electron side
+  if (environment.isDesktop) {
+    console.log('set config', settings);
+    window.apis?.updater
+      .setConfig({
+        autoCheckUpdate: settings.autoCheckUpdate,
+        autoDownloadUpdate: settings.autoDownloadUpdate,
+      })
+      .catch(err => {
```
Comment (joooye34) [inline packages/common/infra/src/atom/settings.ts:80]: can we use sentry to report error here?

File: packages/frontend/component/src/components/app-sidebar/app-updater-button/index.tsx [hunk 3]
```diff
@@ -38,24 +30,141 @@ export function AppUpdaterButtonPure({
...
+              ]()}
+            </span>
+          </div>
+        </div>
+      );
+    }
+
+    function renderDownloadingUpdate() {
+      return (
+        <div className={clsx([styles.updateAvailableWrapper])}>
```
Comment (joooye34) [inline packages/frontend/component/src/components/app-sidebar/app-updater-button/index.tsx]: most clsx is unneccessary

File: packages/frontend/component/src/components/app-sidebar/app-updater-button/index.tsx [hunk 4]
```diff
@@ -38,24 +30,141 @@ export function AppUpdaterButtonPure({
...
+              {t[
+                appQuitting ? 'Loading' : 'com.affine.appUpdater.installUpdate'
+              ]()}
+            </span>
+          </div>
+        </div>
+      );
+    }
+
+    function renderDownloadingUpdate() {
```
Comment (joooye34) [inline packages/frontend/component/src/components/app-sidebar/app-updater-button/index.tsx]: Seems those functions should be split into component and state map. It is so weird to use the inner render function

File: packages/common/infra/src/atom/settings.ts (hunk 1 above)
Comment (pengx17) [reply packages/common/infra/src/atom/settings.ts:6]: Due to some module import side effect sequence issue, there is a case that environment variables have not being initialized yet.

File: packages/common/infra/src/atom/settings.ts (hunk 2 above)
Comment (pengx17) [reply packages/common/infra/src/atom/settings.ts:80]: I think console.error calls are automatically captured by sentry. correct me if im wrong

File: packages/frontend/component/src/components/app-sidebar/app-updater-button/index.tsx (hunk 3 above)
Comment (pengx17) [reply packages/frontend/component/src/components/app-sidebar/app-updater-button/index.tsx]: done

File: packages/common/infra/src/atom/settings.ts (hunk 2 above)
Comment (joooye34) [reply packages/common/infra/src/atom/settings.ts:80]: Sentry will not report console.error, unless using this integration https://docs.sentry.io/platforms/javascript/configuration/integrations/captureconsole/?original_referrer=https%3A%2F%2Fdocs.sentry.io%2Fplatforms%2Fjavascript%2Fguides%2Freact%2Ftroubleshooting%2F%3Foriginal_referrer%3Dhttps%253A%252F%252Fblog.sentry.io%252F.

File: packages/common/infra/src/atom/settings.ts (hunk 1 above)
Comment (joooye34) [reply packages/common/infra/src/atom/settings.ts:6]: ok, but leave some comments to improve those codes?

Comment (joooye34) [reply packages/common/infra/src/atom/settings.ts:6]: Seems those sequence issue should be resolved.

**Review by joooye34 — APPROVED** [review]

File: packages/common/infra/src/atom/settings.ts (hunk 2 above)
Comment (joooye34) [reply packages/common/infra/src/atom/settings.ts:80]: maybe let me improve it later together.

**Pull request merged**